### ISSUE TITLE
MySQL TLS connection support with verify-server-cert=false

### ISSUE THREAD

**Issue by tomaskir** [FreeRADIUS/freeradius-server#2475]
This is a feature request to support TLS connections towards MySQL databases without checking the cert of the MySQL server.

Affected modules: `rlm_sql_mysql`
It should be fairly simple to modify `rlm_sql_mysql.c`  around lines `72`,  and `183`.

From user-facing perspective, modifying `mods-available/sql` like this would be useful:
```
...
        mysql {
                tls {
                        verify_server_certificate = yes
...
```

User could then set `verify_server_certificate = no`, and not set `ca_*` and other options, but still have TSL connections to the MySQL server.

The point of this request is to make it easier and faster to deploy TLS connections towards MySQL.
(of course at the cost of compromising security by not validating the server cert)

Why would this be useful?

This is useful in test and dev environments so certs don't have to be transfered around.
Also useful when orchestrating docker deploys for test/dev purposes and wanting to use TLS but not validate the certs.

Basically any automated temporary deployment (as mentioned, for test/dev) would be much easier and faster having this option.

**Comment by alandekok** [comment]
Sure.  We welcome a patch if it's simple.

**Comment by tomaskir** [comment]
Sadly, I am not a C / C++ developer - the code is simple to understand, but modifying it is a different thing.

I can attempt this, but I will not have a way to test if the build passes or not.
(I don't have a build environment for freeradius setup, nor have I ever set one up)

I can however submit a PR that someone can review and fix any errors.
Are there any tests that need to be looked at / fixed / amended after changing the related code?

**Comment by arr2036** [comment]
Even in MySQL 8.0 there doesn't seem to be an option to do that in their C API.  Have a look through and see if you can spot something I missed... https://dev.mysql.com/doc/refman/8.0/en/c-api-encrypted-connections.html

**Comment by arr2036** [comment]
https://github.com/FreeRADIUS/freeradius-server/commit/82f773810acdedb40354d04a37df964dd8fb5878

Here's something that should work for the master branch.

**Comment by tomaskir** [comment]
Thank you for that, it was enough to get us going.
I have asked a friend who is a C++ dev to help out.

There should now be PR in both `master` and `v3.0.x` branches that have working and tested code.

Config that works:
```
        mysql {
                tls {
                        tls_required = yes
                        tls_check_cert = no
                        tls_check_cert_cn = no
                }
        }
```

It was tested against MariaDB 10.3 with a self-signed cert used for TLS.

**Comment by tomaskir** [comment]
As the PRs were now merged, this can be closed.
